components/locale_canonicalizer/src/provider.rs

```diff
@@ -26,18 +28,29 @@ pub mod key {
     derive(serde::Serialize, serde::Deserialize)
 )]
 #[yoke(cloning_zcf)]
-#[allow(missing_docs)] // TODO(#1027) - Add missing docs.
+/// Alias data for locale canonicalization.
```

Comment:
Issue (here and LieklySubtagsV1): Assume people visiting this docs page do not know how alias substitution works, and they want to add or modify the data.  Re-stating the struct and module name in the documentation is not helpful for those users.  Please add more clarity to let them know what they can do with this struct and how they can customize it for their needs.

Reply:
I don't see a good reason for people to use the LocaleCanonicalizer with data that is not coming from CLDR. In particular, with the way the canonicalization algorithm is specified in the unicode documentation, it is not even guaranteed to terminate with data other that what is present in CLDR. We've also tried to optimize things in the implementation based upon assumptions about what is in the CLDR data, so while using it with other data should hopefully still work, it is not something we test, and likely won't be as performant.

I think that documenting things to the level to which you are asking would be a large undertaking to support a use case that we should not be encouraging.

I'm happy to add some documentation here that explains a bit more about how things work and why people would be better off sticking to what is in CLDR rather than trying to use other data sources here.

Reply:
Adding more clarity about what these rules mean, what they do, and pointing to CLDR for more details sounds like a good direction to me.